tck/features/clauses/match-where/MatchWhere4.feature

```diff
@@ -60,8 +60,8 @@ Feature: MatchWhere4 - Non-Equi-Joins on variables
       """
       MATCH (a), (b)
       WHERE a.id = 0
-        AND (a)-[:T]->(b:TheLabel)
-        OR (a)-[:T*]->(b:MissingLabel)
+        AND NOT isEmpty((a)-[:T]->(b:TheLabel))
+        OR NOT isEmpty((a)-[:T*]->(b:MissingLabel))
```

Comment:
This seems a deliberate decision against `EXISTS { ... }`, which looks like the more straightforward replacement.

```
      MATCH (a), (b)
      WHERE a.id = 0
        AND (a)-[:T]->(b:TheLabel)
        OR (a)-[:T*]->(b:MissingLabel)
        AND EXISTS { (a)-[:T]->(b:TheLabel) }
        OR EXISTS { (a)-[:T*]->(b:MissingLabel) }
      RETURN DISTINCT b
```

Any particular reason or rational behind that?

Reply:
Oh this is nicer, I just didn't realise that you could rewrite it like this.